featuretools/synthesis/encode_features.py

```diff
@@ -65,6 +65,8 @@ def encode_features(feature_matrix, features, top_n=10, include_unknown=True,
         X = feature_matrix.copy()
 
     encoded = []
+    feature_names = [f.get_name() for f in features]
+    extra_columns = [col for col in X.columns if col not in feature_names]
```

Comment:
to make this safer, can we add in a check that all the features you pass in are columns in the feature matrix? with this change, we might silence issues where the feature matrix and features don't match